ext/mbstring/mbstring.c

```diff
@@ -2183,6 +2190,196 @@ PHP_FUNCTION(mb_output_handler)
+        mb_len = string.len>>2; /* eq. string.len / 4 */
+        chunk_len = split_length<<2; /* eq. split_length * 4 */
+    /*
+     * +----------------------------------------------------------------------+
+     * | second scenario: "variable width encodings with length table"        |
+     * +----------------------------------------------------------------------+
+     */
+    } else if (mbfl_encoding->mblen_table != NULL) {
+        char unsigned const *mbtab = mbfl_encoding->mblen_table;
+
+        /* assume that we have 2-bytes characters */
+        array_init_size(return_value, (string.len>>1 + split_length) / split_length);
+
+        while (p < last) { /* split cycle work until the cursor has reached the last byte */
+            char const *chunk_p = p; /* chunk first byte pointer */
+            chunk_len = 0; /* chunk length in bytes */
+            for (uint32_t char_count = 0; char_count < split_length; ++char_count) {
+                char unsigned const m = mbtab[*p]; /* single character length table */
+                chunk_len += m;
+                p += m;
+            }
+            if (p > last) chunk_len -= p - last; /* check if chunk is in bounds */
+            add_next_index_stringl(return_value, chunk_p, chunk_len);
+        }
+        return;
+
+    /*
+     * +----------------------------------------------------------------------+
+     * | third scenario: "else multibyte encodings"                           |
+     * +----------------------------------------------------------------------+
+     */
+    } else {
+        mbfl_convert_filter *filter, *decoder;
+
+        /* assume that we have 2-bytes characters */
+        array_init_size(return_value, (string.len>>1 + split_length) / split_length);
+        p = string.val; /* reset string cursor position */
+
+        /* decoder filter to decode wchar to encoding */
+        mbfl_memory_device device;
+        mbfl_memory_device_init(&device, split_length + 1, 0);
+
+        decoder = mbfl_convert_filter_new(
+                &mbfl_encoding_wchar,
+                string.encoding,
+                mbfl_memory_device_output,
+                0,
+                &device);
+        if (decoder == NULL){
+            mbfl_convert_filter_delete(decoder);
+            RETURN_FALSE; /* something wrong with the filter */
+        }
+
+        /* wchar filter */
+        mbfl_string_init(&result_string); /* mbfl_string to store chunk in the callback */
+        struct mbfl_split_params params = { /* init callback function params structure */
+                .return_value = return_value,
+                .result_string = &result_string,
+                .mb_chunk_length = 0,
+                .split_length = (size_t)split_length,
+                .next_filter = decoder,
+        };
+
+        filter = mbfl_convert_filter_new(
+                string.encoding,
+                &mbfl_encoding_wchar,
+                mbfl_split_output,
+                0,
```

Comment:
This should also be NULL.

Reply:
It is. Fixed here. 
These mistakes should also be fixed everywhere. For example: 
https://github.com/php/php-src/blob/686c29531e4804eda49746fa6e62b9ffe5e0a9fc/ext/mbstring/mbstring.c#L4656